lib/clientv3/globalnetworkpolicy.go

```diff
@@ -48,6 +51,11 @@ func (r globalNetworkPolicies) Create(ctx context.Context, res *apiv3.GlobalNetw
 		return nil, err
 	}
 
+	// Check if it is permitted to create the network policy with the specific alpha feature support.
```

Comment:
I think I'd prefer to move this code into the generic validation (see line 50 above), but I'll defer to @robbrockbank on that one.

Reply:
Probably here: https://github.com/projectcalico/libcalico-go/blob/master/lib/validator/v3/validator.go#L617

Reply:
@casey @robbrockbank I've combined the checks into one place instead of both gnp and np having the same code.
i don't quite understand how to make it part of the validator though. The check needs to reference the alpha_feature configuration flag and it is not obvious to me how to get that context into the validator.
If it's not straight forward then maybe this is something we can handle post merge? wdyt

Reply:
I think it's fine revisiting this later as we don't currently have the infrastructure to do this in the generic validator.

Reply:
Is the problem just that the validator doesn't have the apiconfig available to it?

Reply:
@caseydavenport @robbrockbank The validator does not have apiconfig is the basic issue.
For now let me move this check to the validator as `validateNoServiceAccountRules` and then invoke it from gnp and np.